src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Parser.cs

```diff
@@ -777,16 +779,41 @@ private DictionarySpec CreateDictionarySpec(INamedTypeSymbol type, Location? loc
             {
                 Debug.Assert(!_createdSpecs.ContainsKey(type));
 
+                IMethodSymbol? ctor = GetConstructor(type);
+
                 // Add spec to cache before traversing properties to avoid stack overflow.
-                if (!HasPublicParameterlessCtor(type))
+
+                if (ctor is null)
                 {
                     ReportUnsupportedType(type, Helpers.NeedPublicParameterlessConstructor, location);
                     _createdSpecs.Add(type, null);
                     return null;
                 }
-                ObjectSpec objectSpec = new(type) { Location = location, ConstructionStrategy = ConstructionStrategy.ParameterlessConstructor };
+
+                ConstructionStrategy constructionStrategy = ctor.Parameters.Length is 0 ? ConstructionStrategy.ParameterlessConstructor : ConstructionStrategy.ParameterizedConstructor;
+
+                ObjectSpec objectSpec = new(type)
+                {
+                    Location = location,
+                    ConstructionStrategy = constructionStrategy,
+                };
+
                 _createdSpecs.Add(type, objectSpec);
 
+                Dictionary<string, ParameterSpec>? parameterCache = null;
+                if (constructionStrategy is ConstructionStrategy.ParameterizedConstructor)
+                {
+                    AddToRootConfigTypeCache(MethodSpecifier.Instantiate, objectSpec);
+
+                    parameterCache = new Dictionary<string, ParameterSpec>(StringComparer.OrdinalIgnoreCase);
```

Comment:
>OrdinalIgnoreCase [](http://example.com/codeflow?start=90&length=17)

why this is ignore casing? isn't allowed to have same spelling parameters but different in casing.

Reply:
https://github.com/dotnet/runtime/pull/86365/files#r1196279402.